R/glue.R

```diff
@@ -19,12 +19,21 @@ glue_message <- function(
     params[bool_names] <- lapply(params[bool_names], function(x){x %||% NA}) # nolint
   }
 
-  params[char_names] <- lapply(char_names, function(char_name) {
-    x <- params[[char_name]] %||% "" # convert NULL strings to "" to work with glue
-    chkm8_single_character(x, char_name)
-    x
-  })
+  params[char_names] <- lapply(char_names, clean_char_names, params = params)
 
   ret <- glue::glue_data(params, glue_expression)
   return(ret)
 }
+
+clean_char_names <- function(char_name, params) {
+  x <- params[[char_name]] %||% "" # convert NULL strings to "" to work with glue
+
+  tryCatch({
+    chkm8_single_character(x, char_name)
```

Comment:
this tryCatch setup seems like a fix for a different problem.

Maybe run `x <- as.character(x)` after `x` is defined on line 29? Then call `chkm8_single_character(x, char_name)`

Reply:
it came up because of `test_result`'s num_correct / total_tests calculation since those values are numbers and threw an error. I initially thought of not using `tryCatch`, but I figure this would be a placeholder for some other edge case in the future?

Reply:
Maybe we add in a new thing of starting with `.num_` is numeric? (Like how we did for bools) 

Otherwise remove the `tryCatch` and just use the `error` portion of it.

Reply:
Just realized, `chkm8_single_character` may not be the correct function to use here.
We just want a valid value that can be used in `glue::glue` but glue can take numeric values, so we should'nt just be checking for characters.

we really just want each element in params to be any atmoic of length 1, and not something like a list, right?

Reply:
moving to use `expect_atomic`: https://mllg.github.io/checkmate/reference/checkAtomic.html